review-notifications/src/background.js

```diff
@@ -181,33 +194,75 @@ function filterPullRequests(prData) {
   state.reviewedPR = reviewedPR;
 }
 
-function checkForDiffrences() {
+async function checkForDiffrences() {
+  await getPullRequests();
   chrome.storage.local.get(
-    ['createdPR'],
+    ['createdPR', 'assignedPR', 'mentionedPR', 'reviewedPR', 'followedPR'],
     function(result) {
-      if (state !== undefined && result.createdPR.length > 0)
-        state.createdPR.map(currCreatedPR => {
-          const pr = result.createdPR.filter(
-            resPr => resPr.link === currCreatedPR.link
-          );
-          if (currCreatedPR.updated !== pr[0].updated)
-            findChanges(currCreatedPR, pr[0]);
-        });
+      checkDataFromChromeStorage(result.createdPR, 'createdPR');
+      checkDataFromChromeStorage(result.assignedPR, 'assignedPR');
+      checkDataFromChromeStorage(result.mentionedPR, 'mentionedPR');
+      checkDataFromChromeStorage(result.reviewedPR, 'reviewedPR');
+      updateStateInStorage();
     }.bind(this)
   );
-  updateStateInStorage();
-  getPullRequests();
+}
+function checkDataFromChromeStorage(listOfPRs, stateName) {
+  if (listOfPRs.length > 0)
+    state[stateName].map(currPR => {
+      const pr = listOfPRs.find(resPr => resPr.link === currPR.link);
+      if (pr && currPR.updated !== pr.updated)
+        checkForNewComments(currPR, pr.updated);
+      else if (!pr)
```

Comment:
`pr` is an array or a single element. by doing `!pr` javascript is running `!bool(pr)` under the hood and converting it into a boolean. This can lead to bugs so it's better to do things like these explicitly like `else if (pr.length > 0)`